### ISSUE TITLE
Define what a micro library is

### ISSUE THREAD

**Issue by solnic** [microrb/microrb.com#2]
What's more important - LOC or functionality provided by a given library? I'm leaning towards functionality especially that narrow scope of a library usually comes with a low LOC _anyway_. I wouldn't set any kind of limits because that would be harmful.

Another thing is dependencies - personally I dislike the argument that if a library has dependencies then it's considered as "big". One of the principles I'd love to highlight via microrb is focusing on interfaces and making it easy to integrate small libraries with each other with a clear separation between lower-level tools and higher-level abstractions built  for convenience (think DSLs mostly). This obviously means that I'm in favor of libraries that actually do depend on other libraries.

Thoughts?

**Comment by CrowdHailer** [comment]
Is it easy to measure interface size. perhaps the number of public methods on classes in the namespace. I think I micro library should definitely create objects only in one namespace. so we could use something like

```rb
require 'test-lib'

TestLib.constants.inject(0) do |total, object|
    item = TestLib.const_get(object)
    count = (item.new.methods - Object.method).count
    total + count
end
```

A few issues with the code above but the fact it can be automated has to be worth something.

**Comment by jarmo** [comment]
I didn't read the whole discussion above, but for me a micro library is some library which does one thing and does it well. Like unix tools usually.

I don't think that following any other metrics (like LOC) would add any value. Sometimes having more LOC is better than less, because the code then might contain less magic making it more readable ;)

TLDR; i don't think that you can "detect" micro library via some automated metrics.
